src/lib.rs

```diff
@@ -317,6 +317,26 @@ impl Check {
             .deps(false)
             .silence(!config.is_verbose());
 
+        // If both the current and baseline rustdoc are given explicitely as a file path, we don't need to use the installed rustc, and this check can be skipped.
+        if !(matches!(self.current.source, RustdocSource::Rustdoc(_))
+            && matches!(self.baseline.source, RustdocSource::Rustdoc(_)))
+        {
+            match rustc_version::version() {
+                Ok(rustc_version) => {
+                    let rustc_version_needed = rustc_version::Version::new(1, 65, 0);
+                    if rustc_version < rustc_version_needed {
+                        anyhow::bail!("rustc version is not high enough: >={} needed, got {}\n\nHELP: to run with the latest rustc version, use `rustup update stable && cargo +stable semver-checks ...`", rustc_version_needed, rustc_version);
+                    }
+                }
+                Err(error) => {
+                    config.shell_warn(format_args!(
+                        "while trying to query the current rustc version: {}",
+                        error
+                    ))?;
```

Comment:
This error message text is "implementation-oriented." The user that would see it wouldn't know we were querying the rustc version (or why) so they'll just be confused and won't know what to do. Let's try to make this a user-oriented, actionable error message:
```suggestion
                    config.shell_warn(format_args!(
                        "failed to determine the current rustc version: {}\n\nHELP: to avoid errors please ensure Rust 1.65+ is used",
                        error
                    ))?;
```
or something to that effect. Bonus points if you could pull the `rustc_version_needed` into `config` itself and then somehow use its value to generate the "Rust 1.65+" portion of the error message string.

Reply:
Does pulling the `rustc_version_needed` into `config` bring other benefits? Right now the message is generated by the local `let rustc_version_needed = rustc_version::Version::new(1, 65, 0)`, and it works well.

Reply:
This local `let` is unfortunately not authoritative on which version is actually supported — it's totally possible that we bump the underlying `trustfall_rustdoc` dependency (which is authoritative on the supported rustdoc versions) while forgetting to change this value here.

It's somewhat less likely that we forget it if it's either a global static, or in the config, than a random local variable in the middle of the file. So it's just about improving the probability of us remembering to change it, nothing else :)

Reply:
Got it!